core/redis-kafka-connect/src/main/java/com/redis/kafka/connect/sink/HsetDel.java

```diff
@@ -0,0 +1,33 @@
+package com.redis.kafka.connect.sink;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.springframework.util.CollectionUtils;
+
+import io.lettuce.core.RedisFuture;
+import io.lettuce.core.api.async.BaseRedisAsyncCommands;
+import io.lettuce.core.api.async.RedisHashAsyncCommands;
+import io.lettuce.core.api.async.RedisKeyAsyncCommands;
+import com.redis.spring.batch.writer.operation.AbstractKeyWriteOperation;
+
+public class HsetDel<K, V, T> extends AbstractKeyWriteOperation<K, V, T> {
+
+    private Function<T, Map<K, V>> mapFunction;
+
+    public void setMapFunction(Function<T, Map<K, V>> map) {
+        this.mapFunction = map;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected RedisFuture<Long> execute(BaseRedisAsyncCommands<K, V> commands, T item, K key) {
```

Comment:
custom execute command. if value is null (i.e. map), then the key is deleted with the `del` command. if value is nonnull, then the key and value are inserted into redis with the `hset` command.

Reply:
existing hset command could not handle key deletion hence this custom function